qemu/target-arm/translate-a64.c

```diff
@@ -10984,10 +10985,10 @@ static void disas_a64_insn(CPUARMState *env, DisasContext *s)
     s->pc += 4;
 
     // Unicorn: trace this instruction on request
-    if (env->uc->hook_insn) {
-        struct hook_struct *trace = hook_find(s->uc, UC_HOOK_CODE, s->pc - 4);
-        if (trace)
-            gen_uc_tracecode(tcg_ctx, 4, trace->callback, env->uc, s->pc - 4, trace->user_data);
+    HOOK_FOREACH(env->uc, hook, UC_HOOK_CODE) {
+        if (! HOOK_BOUND_CHECK(hook, s->pc - 4))
+            continue;
+        gen_uc_tracecode(tcg_ctx, 4, hook->callback, env->uc, s->pc - 4, hook->user_data);
```

Comment:
A single hook will be slower because the address bounds check won't happen at JIT time. I think we should just patch the size arguments for all of them.